crates/bevy_color/src/linear_rgba.rs

```diff
@@ -228,6 +279,90 @@ impl From<LinearRgba> for wgpu::Color {
     }
 }
 
+/// All color channels are scaled directly,
+/// but alpha is unchanged.
+///
+/// Values are not clamped.
+impl Mul<f32> for LinearRgba {
+    type Output = Self;
+
+    fn mul(self, rhs: f32) -> Self {
+        Self {
+            red: self.red * rhs,
+            green: self.green * rhs,
+            blue: self.blue * rhs,
+            alpha: self.alpha,
+        }
+    }
+}
+
+impl Mul<LinearRgba> for f32 {
+    type Output = LinearRgba;
+
+    fn mul(self, rhs: LinearRgba) -> LinearRgba {
+        rhs * self
+    }
+}
+
+/// All color channels are scaled directly,
+/// but alpha is unchanged.
+///
+/// Values are not clamped.
+impl Div<f32> for LinearRgba {
+    type Output = Self;
+
+    fn div(self, rhs: f32) -> Self {
+        Self {
+            red: self.red / rhs,
+            green: self.green / rhs,
+            blue: self.blue / rhs,
+            alpha: self.alpha,
+        }
+    }
+}
+
+/// All color channels are added directly,
+/// but alpha is the average of the two alphas.
+impl Add<LinearRgba> for LinearRgba {
+    type Output = LinearRgba;
+
+    fn add(self, rhs: LinearRgba) -> LinearRgba {
+        LinearRgba {
+            red: self.red + rhs.red,
+            green: self.green + rhs.green,
+            blue: self.blue + rhs.blue,
+            alpha: (self.alpha + rhs.alpha) / 2.0,
```

Comment:
I'm really not sure of the best way to handle alpha with color addition and subtraction. Our options:

1. Average: leads to fairly surprising behavior, non-associative.
2. Use the left-hand alpha: fairly surprising, non-commutative.
3. Set the alpha to 1: very surprising, associative and commutative.

Reply:
Another possibility is to do a alpha multiplication, as to say "the alpha will blend in one another". If we have ½ + ¼, this will turn in ⅛. But I'm not sure of this behavior in a add function. I think 1 is the better.

Reply:
In my non-expert opinion, adding colors doesn't make any sense and maybe we shouldn't provide this.

Reply:
Why is alpha singled out?
Couldn't we just treat it as we do with the other components (it gets scaled by * and / and gets added/subtracted by + and -).
That would make the most sense when thinking about colours as an affine (or rather convex) space.
`0.5 * BLACK` is somewhat poorly defined and doesn't represent a colour, but
`0.5 * BLACK + 0.5 * WHITE` makes perfect sense.
Generally, you can take any set of coefficients `a_i >= 0` that sum to 1 and do a convex combination `sum_i a_i * color_i`. I think that is completely reasonable behaviour. I'm not a color-space expert, but I think that all the colour space gamuts we have are convex sets? (Therefore this kind of operation creates valid colours.)

This is also relevant if we want to implement Point on colours and use color curves. The code for curves generally assumes a space closed under convex combinations, with a commutative and associative addition operation (and a scaling operation that distributes over addition and agrees with scalar multiplication (meaning that `(ab) * Point = a * (b * Point)`), at least as far as I remember those are the assumptions made.)

Reply:
We definitely don't want to multiply or divide alpha: multiplying and dividing colors by floats is used for things like HDR and emissivity, and suddenly affecting alpha there will lead to subtle bugs.

In terms of addition and subtraction, one of the things I like about the averaging (and all other methods proposed): we stay bounded between 0 and 1 alpha (assuming inputs within that range). What does an alpha of *2* mean? An alpha of -1?

There is a secret 4th options of course: just panic with non 0 or 1 alpha! That seems particularly nasty though, and likely to slow down this operation.

Generally I think we should punt this question to follow-up work, and tackle it once we can see how colors play with both the curves and the `Animatable` trait.

Reply:
In the linear domain using physical units, adding RGB together makes sense as just "more light". What does adding alpha mean? Should it apply 'over' style alpha blending i.e. when adding colours A and B together, it would produce:
```rust
red: self.red * self.alpha + other.red * (1.0 - self.alpha),
green: self.green * self.alpha + other.green * (1.0 - self.alpha),
blue: self.blue * self.alpha + other.blue * (1.0 - self.alpha),
alpha: self.alpha + other.alpha * (1 - self.alpha),
```
?

Reply:
In the linear domain using physical units, what does alpha mean?

Reply:
In general, I think we should avoid overloading math operators unless they have a clear and unambiguous meaning.

HDR is an important niche, but it's still a niche, and we've already run into the problem that the "meaning" of r, g, and b channels are different in an HDR context than they are in other contexts; common intuitions about color operations, such as clamping, don't necessarily hold. We risk creating confusion if we add a bunch of HDR-specific calculation operations that aren't useful outside of an HDR domain. These would be better done via traits that are clearly marked as being HDR related.